pxr/usd/usd/prim.h

```diff
@@ -2379,8 +2312,138 @@ class UsdPrimSiblingIterator {
     Usd_PrimFlagsPredicate _predicate;
 };
 
-// Typedef iterator range.
-typedef boost::iterator_range<UsdPrimSiblingIterator> UsdPrimSiblingRange;
+/// Forward iterator range of sibling ::UsdPrim s.  This range type contains a
+/// pair of UsdPrimSiblingIterator s, denoting a half-open range of UsdPrim
+/// siblings.  It provides a subset of container-like API, such as begin(),
+/// end(), front(), empty(), etc.
+class UsdPrimSiblingRange {
+public:
+    /// Iterator type.
+    typedef UsdPrimSiblingIterator iterator;
+    /// Const iterator type.
+    typedef UsdPrimSiblingIterator const_iterator;
+    /// Iterator difference type.
+    typedef std::ptrdiff_t difference_type;
+    /// Iterator value_type.
+    typedef iterator::value_type value_type;
+    /// Iterator reference_type.
+    typedef iterator::reference reference;
+
+    UsdPrimSiblingRange() = default;
```

Comment:
I'm curious about having to explicit default the implicit copy/move constructors/assignment. Is this necessary because the templated constructors below create a user-defined copy constuctor when expanded against UsdPrimSiblingRange?
If we get rid of the templated constructor/assignment, can we get rid of these declarations too?